spark2/src/main/java/org/apache/iceberg/spark/source/Reader.java

```diff
@@ -283,7 +284,8 @@ public Statistics estimateStatistics() {
       return new Stats(0L, 0L);
     }
 
-    if (filterExpressions == null || filterExpressions == Expressions.alwaysTrue()) {
```

Comment:
We had a wrong predicate before: we compared a list to an expression.

Reply:
And it was always `null` then?

Reply:
Yeah, it was null

Reply:
thanks for catching this @aokolnychyi